### PR TITLE
[SFN] ASL Grammars Enhancements

### PR DESCRIPTION
**Pull request by MEPalma**
## Motivation
This PR enhances the reliability and maintainability of the ASL parsers and grammars. By ensuring that start productions consume End-of-File (EOF), potential parsing errors are mitigated, leading to a more complete/resilient parsing process. Additionally, the introduction of a format configuration ensures uniformity in code formatting.

## Changes
- Changes in parser grammars and preprocessors to ensure start productions consume EOF
- Added format config to all ANTLR4 grammars

### CODE REVIEW

File: localstack/services/stepfunctions/asl/antlr/ASLIntrinsicParser.g4 [hunk 1]
```diff
@@ -31,19 +31,16 @@ state_fun_name
...
-    ;
+func_arg:
+    STRING                # func_arg_string
+    | INT                 # func_arg_int
+    | NUMBER              # func_arg_float
+    | (TRUE | FALSE)      # func_arg_bool
+    | CONTEXT_PATH_STRING # func_arg_context_path
+    | JSON_PATH_STRING    # func_arg_json_path
+    | states_func_decl    # func_arg_func_decl
+;
```
Comment (joe4dev) [inline localstack/services/stepfunctions/asl/antlr/ASLIntrinsicParser.g4:46]: Are EOF newlines are possible to validate/unify with the formatter?

File: localstack/services/stepfunctions/asl/antlr/ASLLexer.g4 [hunk 2]
```diff
@@ -1,205 +1,185 @@
...
+WAIT     : '"Wait"';
+PARALLEL : '"Parallel"';
+MAP      : '"Map"';
+
+CHOICES  : '"Choices"';
+VARIABLE : '"Variable"';
+DEFAULT  : '"Default"';
+BRANCHES : '"Branches"';
+
+AND                            : '"And"';
```
Comment (joe4dev) [inline localstack/services/stepfunctions/asl/antlr/ASLLexer.g4:42]: just an observation: it's always a trade-off for long blocks: nice readability through indentation but a single line change can update the entire block. Overall, I think the alignment is still a great readability improvement for such grammar specs.

File: localstack/services/stepfunctions/asl/antlr/ASLIntrinsicLexer.g4 [hunk 3]
```diff
@@ -1,84 +1,59 @@
-lexer grammar ASLIntrinsicLexer;
+// $antlr-format alignTrailingComments true, columnLimit 150, maxEmptyLinesToKeep 1, reflowComments false, useTab false
```
Comment (joe4dev) [inline localstack/services/stepfunctions/asl/antlr/ASLIntrinsicLexer.g4:1]: nice ✨  Are these known best practices for *.g4 files or did you come up with these rules?

File: localstack/services/stepfunctions/asl/antlr/runtime/ASLIntrinsicParser.py [hunk 4]
```diff
@@ -117,6 +118,9 @@ def states_func_decl(self):
             return self.getTypedRuleContext(ASLIntrinsicParser.States_func_declContext,0)
 
 
+        def EOF(self):
```
Comment (joe4dev) [inline localstack/services/stepfunctions/asl/antlr/runtime/ASLIntrinsicParser.py:121]: That's the main functional change I guess. It's a bit hard to spot within all the re-formatting 🕵️

**Review by joe4dev — APPROVED** [review]
The formatting improves consistency and readability ✨ 
The functional changes are a bit hard to spot among all the re-formatting 🕵️ 

How is the re-formatting done? It would be helpful to add a `format` target to the StepFunctions Makefile proposed in https://github.com/localstack/localstack/pull/10193

Idea💡 : Maybe, would it be useful to have a pre-commit hook running upon changes within `antlr` to trigger the formatting such that it remains consistent? I guess that depends a bit on how often we expect to update the grammar.

File: localstack/services/stepfunctions/asl/antlr/ASLLexer.g4 (hunk 2 above)
Comment (MEPalma) [reply localstack/services/stepfunctions/asl/antlr/ASLLexer.g4:42]: I totally agree. For the time being the maximum line width is set to 150, it seems to be standard practice in https://github.com/antlr/grammars-v4

File: localstack/services/stepfunctions/asl/antlr/ASLIntrinsicParser.g4 (hunk 1 above)
Comment (MEPalma) [reply localstack/services/stepfunctions/asl/antlr/ASLIntrinsicParser.g4:46]: Unfortunately I don't see an options to achieved this

File: localstack/services/stepfunctions/asl/antlr/ASLIntrinsicLexer.g4 (hunk 3 above)
Comment (MEPalma) [reply localstack/services/stepfunctions/asl/antlr/ASLIntrinsicLexer.g4:1]: Yes, the same are used in the antlr4 grammar repository: https://github.com/antlr/grammars-v4

**Comment by MEPalma** [comment]
Thank you for the review! I agree with you, the next step is to add formatting commands to the Makefile, I will integrate them in https://github.com/localstack/localstack/pull/10193

**Comment by joe4dev** [comment]
sidenote: we usually rebase instead of merge back master to keep the history cleaner, but it does not matter too much with squashed merges

**Pull request merged**